### ISSUE TITLE
Problem when trying to stream from crunchyroll with mpv

### ISSUE THREAD

**Issue by jvschiavo** [yt-dlp/yt-dlp#2074]
I'm having problems when trying to stream from crunchyroll. I noticed the problem yesterday.

Here is what I get when I try to use mpv (I have a symlink youtube-dl > yt-dlp):

```
$ mpv https://www.crunchyroll.com/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083
[ffmpeg] https: HTTP error 403 Forbidden
Failed to open https://www.crunchyroll.com/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083.
[ytdl_hook] 
[ytdl_hook] stack traceback:
[ytdl_hook] 	@ytdl_hook.lua:566: in function 'run_ytdl_hook'
[ytdl_hook] 	@ytdl_hook.lua:725: in function 'fn'
[ytdl_hook] 	mp.defaults:544: in function 'handler'
[ytdl_hook] 	mp.defaults:486: in function 'call_event_handlers'
[ytdl_hook] 	mp.defaults:520: in function 'dispatch_events'
[ytdl_hook] 	mp.defaults:479: in function <mp.defaults:478>
[ytdl_hook] 	[C]: in ?
[ytdl_hook] 	[C]: in ?
[ytdl_hook] Lua error: @ytdl_hook.lua:566: attempt to concatenate local 'err' (a nil value)
client removed during hook handling


Exiting... (Errors when loading file)

```

When trying directly with yt-dlp:

```
yt-dlp -v -F https://www.crunchyroll.com/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083
[debug] Command-line config: ['-v', '-F', 'https://www.crunchyroll.com/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.01 [91f071af6] (zip)
[debug] Python version 3.9.2 (CPython 64bit) - Linux-5.10.0-10-amd64-x86_64-with-glibc2.31
[debug] exe versions: ffmpeg 4.3.3-0, ffprobe 4.3.3-0, rtmpdump 2.4
[debug] Optional libraries: keyring, sqlite
[debug] Proxy map: {}
[debug] [crunchyroll] Extracting URL: https://www.crunchyroll.com/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083
[crunchyroll] 652083: Downloading webpage
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-frFR information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-arME information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-deDE information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-itIT information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-ptBR information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-esLA information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-enUS information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-esES information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[crunchyroll] 652083: Downloading media info
WARNING: [crunchyroll] Unable to download XML: HTTP Error 404: Not Found
[info] Available formats for 652083:
ID                                          EXT RESOLUTION FPS │   TBR PROTO  │ VCODEC        VBR ACODEC    ABR MORE INFO
[...]
```

Btw, I just tried without the -F parameter and it started downloading... I don't know why I can't stream it with mpv like I have been doing. Maybe crunchyroll changed something on their side?

_Originally posted by @jvschiavo in https://github.com/yt-dlp/yt-dlp/issues/2052#issuecomment-998293966_

**Comment by jvschiavo** [comment]
I forked the issue from another one and it didn't offer me a template. I just copied the template from another issue and I'm filling it now:

### Checklist

- [x]  I'm reporting a broken site
- [x]  I've verified that I'm running yt-dlp version 2021.12.01. (update instructions)
- [ ]  I've checked that all provided URLs are alive and playable in a browser*
- [x]  I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x]  I've searched the bugtracker for similar issues including closed ones. DO NOT post duplicates
- [ ]  I've read the guidelines for opening an issue**
- [ ]  I've read about sharing account credentials and I'm willing to share it if required

* Conveniently, the crunchyroll website is not working for me right now... I'l check again later. But weirdly enough, yt-dlp is able to download from it just fine, my problem is when I'm trying to stream it with mpv.
** Oops, I'm gonna read it now.

### Region
Brazil

### Description
Description provided above.

### Verbose log
The verbose output that clearly demonstrates the problem is the one with mpv:

```
$ mpv https://www.crunchyroll.com/pt-br/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083
[ffmpeg] https: HTTP error 403 Forbidden
Failed to open https://www.crunchyroll.com/pt-br/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083.
[ytdl_hook] 
[ytdl_hook] stack traceback:
[ytdl_hook] 	@ytdl_hook.lua:566: in function 'run_ytdl_hook'
[ytdl_hook] 	@ytdl_hook.lua:725: in function 'fn'
[ytdl_hook] 	mp.defaults:544: in function 'handler'
[ytdl_hook] 	mp.defaults:486: in function 'call_event_handlers'
[ytdl_hook] 	mp.defaults:520: in function 'dispatch_events'
[ytdl_hook] 	mp.defaults:479: in function <mp.defaults:478>
[ytdl_hook] 	[C]: in ?
[ytdl_hook] 	[C]: in ?
[ytdl_hook] Lua error: @ytdl_hook.lua:566: attempt to concatenate local 'err' (a nil value)
client removed during hook handling


Exiting... (Errors when loading file)
```
If I'm not mistaken I was able to stream it just fine a few days ago even though I was getting the 2 first lines from the log above.

The log when using yt-dlp:

```
$ yt-dlp -Uv https://www.crunchyroll.com/pt-br/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083
[debug] Command-line config: ['-Uv', 'https://www.crunchyroll.com/pt-br/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.01 [91f071af6] (zip)
[debug] Python version 3.9.2 (CPython 64bit) - Linux-5.10.0-10-amd64-x86_64-with-glibc2.31
[debug] exe versions: ffmpeg 4.3.3-0, ffprobe 4.3.3-0, rtmpdump 2.4
[debug] Optional libraries: keyring, sqlite
[debug] Proxy map: {}
Latest version: 2021.12.01, Current version: 2021.12.01
yt-dlp is up to date (2021.12.01)
[debug] [crunchyroll] Extracting URL: https://www.crunchyroll.com/pt-br/jojos-bizarre-adventure/episode-2-a-letter-from-the-past-652083
[crunchyroll] 652083: Downloading webpage
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-esLA information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-frFR information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-itIT information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-enUS information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-deDE information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-ptBR information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-esES information
[crunchyroll] 652083: Downloading adaptive_hls-audio-jaJP-hardsub-arME information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[crunchyroll] 652083: Downloading media info
WARNING: [crunchyroll] Unable to download XML: HTTP Error 404: Not Found
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 652083: Downloading 1 format(s): adaptive_hls-audio-jaJP-8115-1
[debug] Invoking downloader on "https://pl.crunchyroll.com/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/index-v1-a1.m3u8?Expires=1640257147&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbC5jcnVuY2h5cm9sbC5jb20vZXZzMS8wZTUxNDc4M2MyMmZmMDdkYTM0NDc3OTYxYzBiZDIyOC9hc3NldHMvNTlqdno4aHQybXh1Zjh4XyoubTN1OD8qIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjQwMjU3MTQ3fX19XX0_&Signature=pWg1a2C75YKUjAJyz4f3EVBRdsk889CVp-91Lcno9EMgCjpKEmW0x~UfjrCxueoa2YtyVy-BHeCgYeo3AISs-c1rXUdwakKNo3NNJBVlkaLdCPSK2dyD-~ApRqUZjxEYMe2SxzL5GS54wJyLBACM~t9Gnt3iznEOJ3hc25-~RLFOh4GdEfPuwsibJKnEXXMWrwKJAr5pT2Bu~i4HTlA5v7L9MH6r1eJkstOVNHOKBakAq58oftkIPwTqHdVvXz5ks4gixj-gSWbcx8wSb5fw770Qw4VurdPX8-R3HGuQLLF4e1K87bEuqMsW5g7Le9ZX1MpA5mPAU-7-gmq32WaOxQ__&Key-Pair-Id=APKAJMWSQ5S7ZB3MF5VA&cdn=fy-prod"
[hlsnative] Downloading m3u8 manifest
WARNING: The stream has AES-128 encryption and pycryptodomex is not available; extraction will be delegated to ffmpeg
[download] Destination: JoJo's Bizarre Adventure - Phantom Blood + Battle Tendency Episódio 2 – Uma Carta do Passado [652083].mp4
[debug] ffmpeg command line: ffmpeg -y -loglevel verbose -headers 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3779.0 Safari/537.36
Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-us,en;q=0.5
Cookie: __cf_bm=KKpStiizNOAntvpQsjnm0yQtNXBEuARloDJmnK9gzp4-1640084345-0-AZir2aNOHaEwHhkv1ojyAvYuixgXmN/Og7hVHKAPEJPYCdqNrpuFiKo7dyNeJ0OlOcpapXxQoITwQs/GEYTkX/Tt/HNk2PheQnJfqe9b1JAZ; session_id=7ba76669d146d6d80e4c073dd4d00a7a; c_visitor=fda957c8-c1ae-454f-8f72-86a8f30ca3eb
' -i 'https://pl.crunchyroll.com/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/index-v1-a1.m3u8?Expires=1640257147&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbC5jcnVuY2h5cm9sbC5jb20vZXZzMS8wZTUxNDc4M2MyMmZmMDdkYTM0NDc3OTYxYzBiZDIyOC9hc3NldHMvNTlqdno4aHQybXh1Zjh4XyoubTN1OD8qIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjQwMjU3MTQ3fX19XX0_&Signature=pWg1a2C75YKUjAJyz4f3EVBRdsk889CVp-91Lcno9EMgCjpKEmW0x~UfjrCxueoa2YtyVy-BHeCgYeo3AISs-c1rXUdwakKNo3NNJBVlkaLdCPSK2dyD-~ApRqUZjxEYMe2SxzL5GS54wJyLBACM~t9Gnt3iznEOJ3hc25-~RLFOh4GdEfPuwsibJKnEXXMWrwKJAr5pT2Bu~i4HTlA5v7L9MH6r1eJkstOVNHOKBakAq58oftkIPwTqHdVvXz5ks4gixj-gSWbcx8wSb5fw770Qw4VurdPX8-R3HGuQLLF4e1K87bEuqMsW5g7Le9ZX1MpA5mPAU-7-gmq32WaOxQ__&Key-Pair-Id=APKAJMWSQ5S7ZB3MF5VA&cdn=fy-prod' -c copy -f mp4 'file:JoJo'"'"'s Bizarre Adventure - Phantom Blood + Battle Tendency Episódio 2 – Uma Carta do Passado [652083].mp4.part'
ffmpeg version 4.3.3-0+deb11u1 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
  configuration: --prefix=/usr --extra-version=0+deb11u1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-pocketsphinx --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
[tcp @ 0x55990f26c280] Starting connection attempt to 13.227.127.77 port 443
[tcp @ 0x55990f26c280] Successfully connected to 13.227.127.77 port 443
[hls @ 0x55990f2675c0] Skip ('#EXT-X-ALLOW-CACHE:YES')
[hls @ 0x55990f2675c0] Skip ('#EXT-X-VERSION:5')
[hls @ 0x55990f2675c0] HLS request for url 'https://fy.v.vrv.co/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/seg-1-v1-a1.ts?t=exp=1640257150~acl=/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/*~hmac=0904ea00d108d3f7bd23d12b9a11bbb053a89620db1fa2fe28d18571df3a925c', offset 0, playlist 0
[hls @ 0x55990f2675c0] Opening 'https://fy.v.vrv.co/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/encryption.key?t=exp=1640257150~acl=/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/*~hmac=0904ea00d108d3f7bd23d12b9a11bbb053a89620db1fa2fe28d18571df3a925c' for reading
[tcp @ 0x55990f5ef2c0] Starting connection attempt to 199.232.114.133 port 443
[tcp @ 0x55990f5ef2c0] Successfully connected to 199.232.114.133 port 443
[AVIOContext @ 0x55990f8e15c0] Statistics: 16 bytes read, 0 seeks
[hls @ 0x55990f2675c0] Opening 'crypto+https://fy.v.vrv.co/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/seg-1-v1-a1.ts?t=exp=1640257150~acl=/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/*~hmac=0904ea00d108d3f7bd23d12b9a11bbb053a89620db1fa2fe28d18571df3a925c' for reading
[tcp @ 0x55990f5eeb40] Starting connection attempt to 199.232.114.133 port 443
[tcp @ 0x55990f5eeb40] Successfully connected to 199.232.114.133 port 443
[h264 @ 0x55990f60bb40] Reinit context to 1920x1088, pix_fmt: yuv420p
Input #0, hls, from 'https://pl.crunchyroll.com/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/index-v1-a1.m3u8?Expires=1640257147&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbC5jcnVuY2h5cm9sbC5jb20vZXZzMS8wZTUxNDc4M2MyMmZmMDdkYTM0NDc3OTYxYzBiZDIyOC9hc3NldHMvNTlqdno4aHQybXh1Zjh4XyoubTN1OD8qIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjQwMjU3MTQ3fX19XX0_&Signature=pWg1a2C75YKUjAJyz4f3EVBRdsk889CVp-91Lcno9EMgCjpKEmW0x~UfjrCxueoa2YtyVy-BHeCgYeo3AISs-c1rXUdwakKNo3NNJBVlkaLdCPSK2dyD-~ApRqUZjxEYMe2SxzL5GS54wJyLBACM~t9Gnt3iznEOJ3hc25-~RLFOh4GdEfPuwsibJKnEXXMWrwKJAr5pT2Bu~i4HTlA5v7L9MH6r1eJkstOVNHOKBakAq58oftkIPwTqHdVvXz5ks4gixj-gSWbcx8wSb5fw770Qw4VurdPX8-R3HGuQLLF4e1K87bEuqMsW5g7Le9ZX1MpA5mPAU-7-gmq32WaOxQ__&Key-Pair-Id=APKAJMWSQ5S7ZB3MF5VA&cdn=fy-prod':
  Duration: 00:23:51.60, start: 0.101000, bitrate: 0 kb/s
  Program 0 
    Metadata:
      variant_bitrate : 0
    Stream #0:0: Video: h264 (High), 1 reference frame ([27][0][0][0] / 0x001B), yuv420p(left), 1920x1080 (1920x1088) [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 90k tbn, 47.95 tbc
    Metadata:
      variant_bitrate : 0
    Stream #0:1: Audio: aac (LC) ([15][0][0][0] / 0x000F), 44100 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 0
    Stream #0:2: Data: timed_id3 (ID3  / 0x20334449)
    Metadata:
      variant_bitrate : 0
Output #0, mp4, to 'file:JoJo's Bizarre Adventure - Phantom Blood + Battle Tendency Episódio 2 – Uma Carta do Passado [652083].mp4.part':
  Metadata:
    encoder         : Lavf58.45.100
    Stream #0:0: Video: h264 (High), 1 reference frame (avc1 / 0x31637661), yuv420p(left), 1920x1080 (0x0) [SAR 1:1 DAR 16:9], q=2-31, 23.98 fps, 23.98 tbr, 90k tbn, 90k tbc
    Metadata:
      variant_bitrate : 0
    Stream #0:1: Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp
    Metadata:
      variant_bitrate : 0
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (copy)
Press [q] to stop, [?] for help
Automatically inserted bitstream filter 'aac_adtstoasc'; args=''
[AVIOContext @ 0x55990f60a040] Statistics: 788472 bytes read, 0 seeks
[hls @ 0x55990f2675c0] HLS request for url 'https://fy.v.vrv.co/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/seg-2-v1-a1.ts?t=exp=1640257150~acl=/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/*~hmac=0904ea00d108d3f7bd23d12b9a11bbb053a89620db1fa2fe28d18571df3a925c', offset 0, playlist 0
[hls @ 0x55990f2675c0] Opening 'crypto+https://fy.v.vrv.co/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/seg-2-v1-a1.ts?t=exp=1640257150~acl=/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/*~hmac=0904ea00d108d3f7bd23d12b9a11bbb053a89620db1fa2fe28d18571df3a925c' for reading
[tcp @ 0x55990f948840] Starting connection attempt to 199.232.114.133 port 443
[tcp @ 0x55990f948840] Successfully connected to 199.232.114.133 port 443
^CNot writing 'clli' atom. No content light level info.
Not writing 'mdcv' atom. Missing mastering metadata.
frame=   48 fps=0.0 q=-1.0 Lsize=     747kB time=00:00:01.99 bitrate=3066.4kbits/s speed=5.88x    
video:713kB audio:32kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.282736%
Input file #0 (https://pl.crunchyroll.com/evs1/0e514783c22ff07da34477961c0bd228/assets/59jvz8ht2mxuf8x_1909827.mp4/index-v1-a1.m3u8?Expires=1640257147&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbC5jcnVuY2h5cm9sbC5jb20vZXZzMS8wZTUxNDc4M2MyMmZmMDdkYTM0NDc3OTYxYzBiZDIyOC9hc3NldHMvNTlqdno4aHQybXh1Zjh4XyoubTN1OD8qIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjQwMjU3MTQ3fX19XX0_&Signature=pWg1a2C75YKUjAJyz4f3EVBRdsk889CVp-91Lcno9EMgCjpKEmW0x~UfjrCxueoa2YtyVy-BHeCgYeo3AISs-c1rXUdwakKNo3NNJBVlkaLdCPSK2dyD-~ApRqUZjxEYMe2SxzL5GS54wJyLBACM~t9Gnt3iznEOJ3hc25-~RLFOh4GdEfPuwsibJKnEXXMWrwKJAr5pT2Bu~i4HTlA5v7L9MH6r1eJkstOVNHOKBakAq58oftkIPwTqHdVvXz5ks4gixj-gSWbcx8wSb5fw770Qw4VurdPX8-R3HGuQLLF4e1K87bEuqMsW5g7Le9ZX1MpA5mPAU-7-gmq32WaOxQ__&Key-Pair-Id=APKAJMWSQ5S7ZB3MF5VA&cdn=fy-prod):
  Input stream #0:0 (video): 48 packets read (730325 bytes); 
  Input stream #0:1 (audio): 87 packets read (32931 bytes); 
  Input stream #0:2 (data): 0 packets read (0 bytes); 
  Total: 135 packets (763256 bytes) demuxed
Output file #0 (file:JoJo's Bizarre Adventure - Phantom Blood + Battle Tendency Episódio 2 – Uma Carta do Passado [652083].mp4.part):
  Output stream #0:0 (video): 48 packets muxed (730325 bytes); 
  Output stream #0:1 (audio): 87 packets muxed (32931 bytes); 
  Total: 135 packets (763256 bytes) muxed
[AVIOContext @ 0x55990f62c180] Statistics: 2 seeks, 6 writeouts
[AVIOContext @ 0x55990f870b00] Statistics: 167920 bytes read, 0 seeks
[AVIOContext @ 0x55990f299880] Statistics: 209406 bytes read, 0 seeks
Exiting normally, received signal 2.

ERROR: Interrupted by user
```

**Comment by pukkandan** [comment]
If it is working in yt-dlp, there is nothing for us to do. You should report it to mpv devs instead

**Comment by jvschiavo** [comment]
> If it is working in yt-dlp, there is nothing for us to do. You should report it to mpv devs instead

I see. I thought the problem could be related to yt-dlp since it seems that mpv uses yt-dlp (and maybe ffmpeg too?) to stream from the website. The lines with problems clearly indicate something with the yt-dlp ([ytdl_hook] Lua error: @ytdl_hook.lua:566: attempt to concatenate local 'err' (a nil value) client removed during hook handling).

Even if that is not the case, I thought it would be useful to post it here since other people might be getting the same problem. I'll check if mpv or ffmpeg projects have a github page.

Thanks for your awesome work!

**Comment by pukkandan** [comment]
https://github.com/mpv-player/mpv/issues

**Comment by nyuszika7h** [comment]
> The lines with problems clearly indicate something with the yt-dlp ([ytdl_hook] Lua error: @ytdl_hook.lua:566: attempt to concatenate local 'err' (a nil value) client removed during hook handling).

ytdl_hook != youtube-dl/yt-dlp itself. It's a bug in their code that handles communicating with yt-dl(p).

**Comment by jvschiavo** [comment]
> > The lines with problems clearly indicate something with the yt-dlp ([ytdl_hook] Lua error: @ytdl_hook.lua:566: attempt to concatenate local 'err' (a nil value) client removed during hook handling).
> 
> ytdl_hook != youtube-dl/yt-dlp itself. It's a bug in their code that handles communicating with yt-dl(p).

I see, thanks for letting me know.

By the way, I have an update:

I asked a friend of mine on Manjaro and she didn't have the same problem, she could stream using mpv just fine (her mpv is 2 major versions ahead of mine, 0.34). I also asked another friend on Mint and he got the same problem I have (his mpv is some versions before mine because stable Mint is based on Debian oldstable right now I think). So maybe this is something Debian-related?

I can't really find a way to isolate the issue, I checked with `sudo debsums -a mpv` and every file is "OK". According to my /var/log/apt/history.log I have not upgraded mpv once in my system, it is the same version since I've installed it:
```
Start-Date: 2021-12-12  04:08:10
Commandline: apt install mpv
Requested-By: user (1000)
Install: python3-pyxattr:amd64 (0.7.2-1+b1, automatic), mpv:amd64 (0.32.0-3), ...
[...]
```
It is the same version that was working before just fine. Maybe someone else is having the same problem. If I find a solution I may update this issue once again.

**Comment by jvschiavo** [comment]
I've found the culprit. It was a problem between the chair and the keyboard. I checked my mpv.conf file and it was missing a `/best` string on my ytdl-raw-options line. I removed it earlier and forgot to put it back, and that was causing all my problems.

The line in my mpv.conf for reference:
`ytdl-raw-options=format="((137/bestvideo)+(251/bestaudio))/best",write-sub=,write-auto-sub=,sub-lang=[enUS,en-US,en]
`
